### PR TITLE
goku: 0.3.6 -> 0.3.11

### PR DESCRIPTION
**Pull request by hlolli**
###### Motivation for this change

Currently goku depends on joker in an impure fashion. The only way to patch this problem away was to build from sources. Which also fixes impure call to watchexec.

###### Things done

Added generated clojure deps file.
Build from sources with graalvm.
Added myself as maintainer.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
Could someone with a mac review?

File: pkgs/os-specific/darwin/goku/default.nix [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
+          $out/bin/goku >> $out/bin/gokuw
+    chmod +x $out/bin/gokuw
   '';
 
   meta = with lib; {
     description = "Karabiner configurator";
     homepage = "https://github.com/yqrashawn/GokuRakuJoudo";
     license = licenses.gpl3;
-    maintainers = [ maintainers.nikitavoloboev ];
+    maintainers = [ maintainers.nikitavoloboev maintainers.hlolli ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/goku/default.nix:51]:
```suggestion
    maintainers = with maintainers; [ nikitavoloboev hlolli ];
```

File: pkgs/os-specific/darwin/goku/default.nix [hunk 2]
```diff
@@ -1,29 +1,54 @@
...
 
   installPhase = ''
-    chmod +x goku
-    chmod +x gokuw
     mkdir -p $out/bin
-    cp goku $out/bin
-    cp gokuw $out/bin
+    cp karabiner_configurator.core $out/bin/goku
+    # https://github.com/yqrashawn/GokuRakuJoudo/blob/a9016fe1ffd700365b1929a87a15c90132903640/.github/workflows/build-and-release.yaml#L34-L38
+    touch $out/bin/gokuw
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/goku/default.nix:40]:
```suggestion
```

File: pkgs/os-specific/darwin/goku/default.nix [hunk 3]
```diff
@@ -1,29 +1,54 @@
...
-    sha256 = "1pss1k2slbqzpfynik50pdk4jsaiag4abhmr6kadmaaj18mfz7ai";
+  src = fetchFromGitHub {
+    owner = "yqrashawn";
+    repo = "GokuRakuJoudo";
+    rev = "v${version}";
+    sha256 = "0j9lk194a4fva866c3q6nhblcb9zfl6nz0gb6zymnsalkp56pfzn";
   };
 
-  sourceRoot = ".";
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/goku/default.nix:21]:
```suggestion
  postPatch = ''
```

File: pkgs/os-specific/darwin/goku/goku.deps.nix [hunk 4]
```diff
@@ -0,0 +1,1252 @@
+# generated by clj2nix-1.0.7
+{ pkgs }:
+
+let repos = [
+        "https://repo1.maven.org/maven2/"
+        "https://repo.clojars.org/" ];
+  in rec {
+      fetchmaven = pkgs.fetchMavenArtifact;
+      makePaths = {extraClasspaths ? null}:
+        (pkgs.lib.concatMap
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/goku/goku.deps.nix:10]:
```suggestion
{ lib, fetchMavenArtifact }:

let repos = [
        "https://repo1.maven.org/maven2/"
        "https://repo.clojars.org/" ];
  in rec {
      fetchmaven = fetchMavenArtifact;
      makePaths = {extraClasspaths ? null}:
        (lib.concatMap
```

File: pkgs/os-specific/darwin/goku/default.nix [hunk 5]
```diff
@@ -1,29 +1,54 @@
-{lib, stdenv, fetchurl }:
+{ lib, stdenv, pkgs, fetchFromGitHub, graalvm, clojure, joker,
+  watchexec, runtimeShell, Foundation, zlib }:
 
-stdenv.mkDerivation rec {
+let cljdeps = import ./goku.deps.nix { inherit pkgs; };
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/goku/default.nix:4]:
```suggestion
let cljdeps = import ./goku.deps.nix { inherit lib, fetchMavenArtifact; };
```

File: pkgs/os-specific/darwin/goku/default.nix [hunk 6]
```diff
@@ -1,29 +1,54 @@
-{lib, stdenv, fetchurl }:
+{ lib, stdenv, pkgs, fetchFromGitHub, graalvm, clojure, joker,
+  watchexec, runtimeShell, Foundation, zlib }:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/goku/default.nix:2]:
```suggestion
{ lib, stdenv, fetchMavenArtifact, fetchFromGitHub, graalvm, clojure, joker
, watchexec, runtimeShell, Foundation, zlib }:
```

Comment (hlolli) [reply pkgs/os-specific/darwin/goku/default.nix]: fixed

**Comment by hlolli** [comment]
The build is broken because of broken watchexec. Older versions work, the newer ones aren't compiling on aarch64-darwin. So this change is blocked by watchexec.

**Comment by wegank** [comment]
Closed by #168863.

**Pull request closed without merging**